cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/validator/CompileValidatorImpl.java

```diff
@@ -67,9 +68,12 @@
         Set<Executable> flowReferences = new HashSet<>();
 
         for (Step step : steps) {
-            Executable reference = dependencies.get(step.getRefId());
-            errors.addAll(validateStepAgainstItsDependency(flow, step, dependencies));
-            flowReferences.add(reference);
+            // validate all steps, except external steps, that are not in the dependencies list
+            if (!(step instanceof ExternalStep)) {
```

Comment:
Maybe `Step` could be an interface and have a method `getType` that returns whether it's `External` or `Regular` (enum). This would avoid the `instanceof` check. (this comment is regarding all instanceof checks of this type)

Reply:
I don't actually see what is the advantage of having a Step interface with two implementations and a getType method? I think that having an extra method to determine whether a step is external or regular might be redundant, when we already know that external steps have a different type then regular steps.
However, it is possible, in this situation to have a polymorphic method requiresValidation and return true or false depending on the case.

Reply:
fair point, then as you said, maybe a method `requiresValidation`?

Reply:
Will do it!